makefiles/common.mk

```diff
@@ -185,8 +185,8 @@ ifeq ($(THEOS_PACKAGE_SCHEME),)
 endif
 _THEOS_INTERNAL_SEARCHPATHS += \
 	$(TARGET_PRIVATE_FRAMEWORK_PATH) \
-	$(THEOS_VENDOR_LIBRARY_PATH)/$(THEOS_TARGET_NAME)/$(or $(THEOS_PACKAGE_SCHEME),rootful) \
-	$(THEOS_LIBRARY_PATH)/$(THEOS_TARGET_NAME)/$(or $(THEOS_PACKAGE_SCHEME),rootful)
+	$(THEOS_VENDOR_LIBRARY_PATH)/$(THEOS_TARGET_NAME)/$(THEOS_PACKAGE_SCHEME) \
+	$(THEOS_LIBRARY_PATH)/$(THEOS_TARGET_NAME)/$(THEOS_PACKAGE_SCHEME)
```

Comment:
Previously, if `THEOS_TARGET_NAME = iphone`, `THEOS_PACKAGE_SCHEME = `, the path added would be `.../iphone/rootful`. With this change, `.../iphone/` would be added to the search paths.
I think this is unexpected.

I think instead, this should be inside of a `ifneq ($(THEOS_PACKAGE_SCHEME),)` (or simply an `else` in this case, since the block above is `ifeq ($(THEOS_PACKAGE_SCHEME),)`) to match the other sites in this PR.

Reply:
Good catch, thanks! In 04173ab0e805f5f506773653f8adf4cdb99e1310 I made the library search paths mutually exclusive and added an addition check for the supplied THEOS_PACKAGE_SCHEME as invalid schemes would then fail to link libroot_oldabi.a due to the missing search paths.